src/array/cuda/array_index_select.cu

```diff
@@ -80,7 +81,7 @@ DType IndexSelect(NDArray array, int64_t index) {
   device->CopyDataFromTo(
       static_cast<DType*>(array->data) + index, 0, reinterpret_cast<DType*>(&ret), 0,
       sizeof(DType), array->ctx, DLContext{kDLCPU, 0},
-      array->dtype, nullptr);
+      array->dtype, thr_entry->stream);
```

Comment:
How about omitting stream here?

Reply:
Sure. I will get rid of the extra `thr_entry->stream`